sql/mysqld.cc

```diff
@@ -6316,6 +6529,9 @@ void handle_connections_sockets()
     create_new_thread(thd);
     set_current_thd(0);
   }
+#ifdef HAVE_SYSTEMD
+  sd_notify(0, "STOPPING=1");
```

Comment:
I did look a bit for the right spot. Some of the background threads might work but they generally depend on enabled features and I thought once you add the WATCHDOG=1 there what functionality of the DBMS are you really ensuring is available? Once I thought about that I thought that the availability test didn't reflect enough DBMS functionality to be useful. If you can think of a good spot happy to add it there.